src/ResourceManager/Automation/Commands.Automation/Common/AutomationPSClient.cs

```diff
@@ -1657,42 +1657,77 @@ private Azure.Management.Automation.Models.HybridRunbookWorkerGroup TryGetHybrid
         private IDictionary<string, string> ProcessRunbookParameters(string resourceGroupName, string automationAccountName, string runbookName,
             IDictionary parameters)
         {
+            Runbook runbook = null;
```

Comment:
This new scenario should have automated test coverage.  We will need to engage the feature team to add auotmated test coverage for this scenario.

Reply:
Are the automated tests part of the project? Just did a quick search through all files and I don't see any test file with references to "ProcessRunbookParameters"

Reply:
The `ScenarioTests` folder in the `Automation.Test` project folder has a lot of tests in it. The tests with `CheckIn` on them are ran when a PR is created. We have some docs here on it: https://github.com/Azure/azure-powershell/blob/master/documentation/testing-docs/using-azure-test-framework.md

Reply:
Checking at the tests, is there any reason why tests for AZ module are using AzureRM cmdlets (I know it might just be the aliases) but curious. Also, in order to add this test, I might need to also modify some other cmdlets which currently don't "fully" support Python2 runbooks which it would change the PR title. Any issues with that?

Reply:
The tests are running the Az cmdlets. The test framework runs `Enable-AzureRmAlias` prior to running tests. Brand new cmdlets won't have a mapping (from AzureRm name to Az name) but you're not making new cmdlets. So, you can continue to use the AzureRm prefix.

I'm confused about your last statement. 
> Also, in order to add this test, I might need to also modify some other cmdlets which currently don't "fully" support Python2 runbooks which it would change the PR title. Any issues with that?

This PR is intended to allow Python 2 runbooks, right? If you cannot run tests with the current changes, then users cannot run Python 2 runbooks with the current changes, correct?

Reply:
Sort of..... because. as part of the testing, you do the environment setup which basically Creates, run and destroys a runbook. However, New-AzAutomationRunbook doesn't support Type = "Python2" but Import-AzAutomationRunbook does. So, in order to have the tests written, I need to extend New-AzAutomationRunbook to support it too.  Makes sense?

Reply:
My changes are to Start-AzAutomationRunbook which doesn't care how that runbook got published

Reply:
That makes sense. It would be beneficial to be able to `New` a Python2 runbook as well as `Start` one, right? If this makes sense, go ahead an implement it. Updating the PR title is fine too.

Reply:
Done, added the test case for starting Python2 Runbooks. As well as extended New-AzAutomationRunbook to support Python2 as type (most of the heavy work was done since Import-AzAutomationRunbook was already supporting it)